src/components/BarChart/index.jsx

```diff
@@ -0,0 +1,68 @@
+import React, { useEffect, useState, useMemo } from "react";
+import { Bar } from "react-chartjs-2";
+import { generateRandomColors, createColorsWithOpacity } from "../../shared/utils";
+
+const CHART_COUNT = 8;
+
+const BarChart = ({ data }) => {
+    const colors = useMemo(() => generateRandomColors(CHART_COUNT), []);
```

Comment:
set initial state using useEffect hook not useMemo

Reply:
I can't, watch this part you will fully understand